bcdoc/restdoc.py

```diff
@@ -98,3 +100,109 @@ def remove_last_doc_string(self):
         if self._last_doc_string is not None:
             start, end = self._last_doc_string
             del self._writes[start:end]
+
+
+class DocumentStructure(ReSTDocument):
+    def __init__(self, name, event_emitter, section_names=None, target='man'):
+        """Provides a Hierarichial structure to a ReSTDocument
+
+        You can write to it similiar to as you can to a ReSTDocument but
+        has an innate structure for more orginaztion and abstraction.
+
+        :param name: The name of the document
+        :param event_emitter: An event emitter used to emit events
+        :param section_names: A list of sections to be included
+            in the document.
+        :parma target: The target documentation of the Document structure
+        """
+        super(DocumentStructure, self).__init__(target=target)
+        self._name = name
+        self._event_emitter = event_emitter
+        self._structure = OrderedDict()
+        self._path = [self._name]
+        if section_names is not None:
+            self._generate_structure(section_names)
+
+    @property
+    def name(self):
+        """The name of the document structure"""
+        return self._name
+
+    @property
+    def path(self):
+        """
+        A list of where to find a particular document structure in the
+        overlying document structure.
+        """
+        return self._path
+
+    @path.setter
+    def path(self, value):
+        self._path = value
+
+    def _generate_structure(self, section_names):
+        for section_name in section_names:
+            self.add_new_section(section_name)
+
+    def add_new_section(self, name):
+        """Adds a new section to the current document structure
+
+        This document structure will be considered a section to the
+        current document structure but will in itself be an entirely
+        new document structure that can be written to and have sections
+        as well
+
+        :param name: The name of the section.
+        :rtype: DocumentStructure
+        :returns: A new document structure to add to but lives as a section
+            to the document structure it was instantiated from.
+        """
+        # Add a new section
+        section = DocumentStructure(
```

Comment:
You can use `self.__class__` so it will work with subclasses and won't have to be updated if the name changes.